tasks/update-cnb-dependency/dependencies.go

```diff
@@ -22,14 +22,25 @@ type Dependency struct {
 	Version      string   `toml:"version"`
 }
 
-func (deps Dependencies) MergeWith(newDeps Dependencies) (Dependencies, error) {
+func (deps Dependencies) Update(flags Flags, dep Dependency, depsToAdd Dependencies) (Dependencies, error) {
```

Comment:
`Dependencies` now has five public methods but only one (`Update`) is used outside of the struct. It's also one of the only methods without tests. Ideally, all of the other methods would be private, but that would mean we'd have to combine all of the test cases which we were trying to avoid. Maybe this method could be put somewhere else and tested? Or the caller could call the functions on `Dependencies` themselves?

Reply:
I'm not sure if my creation of the `UpdateDependenciesWith()` method in buildpack_toml helped or not. I'd love to hear your thoughts